### ISSUE TITLE
Bug report: ')' after argument in arrow function expression 

### ISSUE THREAD

**Issue by hhhr** [jslint-org/jslint#379]
**Describe the bug**
1. A clear and concise description of what the bug is.
In the Code snippet jslint complains about the ')' after the argument x. It wants it to match to the '(' of the .forEach, I guess.
But if I leave it out it complains about not wrapping the argument x in parantheses.
2. Jslint edition:
v2021.11.20
3. Code snippet (if applicable):

```javascript
/*jslint devel*/
const a = [];
a.forEach((x) => delete x.unwanted_key);
```

**Expected behavior**

I expect either jslint complains about me using arrow functions instead of a function declaration
or jslint does not complain at all.

**Comment by kaizhu256** [comment]
thx for reporting.  this seems to be a legacy-bug (pre v2020.11.6) that might need a parser update.  i'll take a closer look at it this weekend.

**Comment by kaizhu256** [comment]
should be fixed in pr #384, with new warning forbidding [side-effect] statements within fat-arrow (jslint expect fat-arrows to only contain expressions)

jslint still warns against your code, but warning is a bit clearer:

```
1. [JSLint was unable to finish] Unexpected 'delete' after '=>'.
```

the new warning applies to following fat-arrow examples where it encounters a statement keyword:

```js
(x) => async ...
(x) => await ...
(x) => break ...
(x) => const ...
(x) => continue ...
(x) => debugger ...
(x) => delete ...
(x) => do ...
(x) => export ...
(x) => for ...
(x) => function ...
(x) => if ...
(x) => import ...
(x) => let ...
(x) => return ...
(x) => switch ...
(x) => throw ...
(x) => try ...
(x) => var ...
(x) => while ...
(x) => with ...
```

closing issue as fixed, but feel free to re-open if you think there's a better-way to deal with the cryptic warnings.

**Comment by hhhr** [comment]
thank you !
